### PR TITLE
CATROID-507 added AssertUserListBrick

### PR DESCRIPTION
**Pull request by Koell**
added abstract MultiUserListBrick which enables multiple UserLists
added functionality of AssertUserListAction which compares two UserLists
adjusted AssertEqualError message
added abstract AssertAction
added new test style so that it can be easily adjusted

*Please enter a short description of your pull request and add a reference to the Jira ticket.*

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/BroadcastMessageBrick.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public View getView(Context context) {
 	}
 
 	@Override
-	public void onNewOptionSelected() {
```
Comment (PlantinumX) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/BroadcastMessageBrick.java:80]: why are you changing also the Broadcastbrick

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/CloneBrick.java [hunk 2]
```diff
@@ -70,16 +70,16 @@ public View getView(Context context) {
 	}
 
 	@Override
-	public void onNewOptionSelected() {
+	public void onNewOptionSelected(Integer spinnerId) {
 	}
 
 	@Override
-	public void onStringOptionSelected(String string) {
+	public void onStringOptionSelected(Integer spinnerId, String string) {
```
Comment (PlantinumX) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/CloneBrick.java:77]: Clonebrick as well

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/BroadcastMessageBrick.java (hunk 1 above)
Comment (Koell) [reply catroid/src/main/java/org/catrobat/catroid/content/bricks/BroadcastMessageBrick.java:80]: it necessary to change the basefunction to enable the identification of multiplible dropdown's

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/CloneBrick.java (hunk 2 above)
Comment (Koell) [reply catroid/src/main/java/org/catrobat/catroid/content/bricks/CloneBrick.java:77]: the same reason as above applies to every similar codechange in different files that added the id as a parameter

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/CategoryBricksFactory.java [hunk 3]
```diff
@@ -667,28 +666,12 @@
...
 		assertionsBrickList.add(new TapAtBrick());
-
 		assertionsBrickList.add(new FinishStageBrick());
 
-		for (Scene scene : ProjectManager.getInstance().getCurrentProject().getSceneList()) {
-			for (Sprite sprite : scene.getSpriteList()) {
-				for (Script script : sprite.getScriptList()) {
-					for (Brick brick : script.getBrickList()) {
-						if (brick instanceof AssertEqualsBrick) {
-							assertionsBrickList.remove(assertEqualsBrick);
```
Comment (PlantinumX) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/CategoryBricksFactory.java:685]: why removing this lines

Comment (Koell) [reply catroid/src/main/java/org/catrobat/catroid/ui/fragment/CategoryBricksFactory.java:685]: to enable multiple assertBricks inside one project without the current workaround

**Comment by PlantinumX** [comment]
What about increasing language version and Xstream?

**Comment by Koell** [comment]
> 
> 
> What about increasing language version and Xstream?

language version was fixed, it might have been overwritten by the current brick additions,
what about XStream?

**Comment by Koell** [comment]
added UserDataBrick instead of MultiUserListBrick, this abstract Brick enables it to have multiple UserLists and UserVariables, that are defined similar to a new Formula in a FormulaBrick

**Comment by wslany** [comment]
Thanks!

**Pull request merged**